src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -70,7 +71,20 @@ private void ResetSolutionSettings()
 
             try
             {
-                _solutionSettings = Settings.LoadDefaultSettings(root, configFileName: null, machineWideSettings: MachineWideSettings);
+                // This is a performance optimization.
+                // The solution load/unload events are called in the UI thread and are used to reset the settings.
+                // In some cases there's a synchronous dependency between the invocation of the Solution event and the settings being reset.
+                // In the open PM UI scenario (no restore run), there is an asynchronous invocation of this code path. This changes ensures that
+                // the synchronus calls that come after the asynchrnous calls don't do duplicate work.
+                // That however is not the case for solution close and  same session close -> open events. Those will be on the UI thread.
+                if (!EqualityUtility.EqualsWithNullCheck(root, _solutionSettings?.Item1))
```

Comment:
I don't understand why you need `EqualityUtility.EqualsWithNullCheck(…)`.  Why not simply:

    if (!string.Equals(root, _solutionSettings?.Item1)

`string.Equals(…)` already has a null check.